packages/core/admin/admin/src/pages/SettingsPage/pages/AuditLogs/ListView/ModalDialog/index.js

```diff
@@ -1,29 +1,60 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
+import { useQuery } from 'react-query';
 import { ModalLayout, ModalHeader, ModalBody } from '@strapi/design-system/ModalLayout';
 import { Breadcrumbs, Crumb } from '@strapi/design-system/Breadcrumbs';
+import { Loader } from '@strapi/design-system/Loader';
 import { Grid } from '@strapi/design-system/Grid';
 import { Box } from '@strapi/design-system/Box';
+import { Flex } from '@strapi/design-system/Flex';
 import { Typography } from '@strapi/design-system/Typography';
-import { pxToRem } from '@strapi/helper-plugin';
+import { pxToRem, useNotification } from '@strapi/helper-plugin';
 import getDefaultMessage from '../utils/getActionTypesDefaultMessages';
 import useFormatTimeStamp from '../hooks/useFormatTimeStamp';
 import ActionItem from './ActionItem';
+import { useFetchClient } from '../../../../../../hooks';
+
+const ModalDialog = ({ onClose, logId }) => {
+  const { get } = useFetchClient();
+  const toggleNotification = useNotification();
+
+  const fetchAuditLog = async (id) => {
+    const { data } = await get(`/admin/audit-logs/${id}`);
+
+    return data;
+  };
+
+  const { data: { date, user, action, payload } = {}, status } = useQuery(
+    ['audit-log', logId],
+    () => fetchAuditLog(logId),
+    {
+      onError() {
+        toggleNotification({
+          type: 'warning',
+          message: { id: 'notification.error', defaultMessage: 'An error occured' },
+        });
+        onClose();
+      },
+    }
+  );
 
-const ModalDialog = ({ onToggle, data: { date, user, action } }) => {
   const { formatMessage } = useIntl();
   const formatTimeStamp = useFormatTimeStamp();
-  const formattedDate = formatTimeStamp(date);
 
-  return (
-    <ModalLayout onClose={onToggle} labelledBy="title">
-      <ModalHeader>
-        <Breadcrumbs label={formattedDate}>
-          <Crumb>{formattedDate}</Crumb>
-        </Breadcrumbs>
-      </ModalHeader>
-      <ModalBody>
+  const formattedDate = status === 'success' ? formatTimeStamp(date) : '';
+
+  const getModalBody = () => {
```

Comment:
I guess this makes sense but usually when I see this kind of pattern it just feels like it should be a component (maybe `ModalBodyContent` co-located). Otherwise I think a ternary would work but I get the feeling you were trying to avoid that.

Also for the name of this component `ModalDialog`, is a dialog a type of Modal? I always thought they were more or less the same thing

Reply:
yeah the idea was to avoid a long ternary, you're right a component is better